android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/AuthenticationManager.java

```diff
@@ -193,11 +205,25 @@ public Completable resendEmailVerification(@NonNull final String email) {
     public Completable signOut() {
         logger.debug("signOut called");
 
+        final String email;
+
+        SignIn signIn = accountDAO.getSignIn();
+        if (signIn != null) {
+            email = signIn.getEmail();
+        } else {
+            email = null;
+            logger.debug("Did not find saved SignIn credentials prior to calling sign out API");
+        }
+
         return RxUtils.toBodySingle(authenticationApi.signOut())
                 .doOnSuccess(message -> {
                     accountDAO.setSignIn(null);
                     accountDAO.setUserSessionInfo(null);
                     accountDAO.setStudyParticipant(null);
+
+                    for (AuthenticationEventListener listener : listeners) {
+                        listener.onSignedOut(email);
```

Comment:
It's usually not the best practice to call listeners on a background thread.  Can we post these to the main thread?